protocols/gossipsub/src/protocol.rs

```diff
@@ -222,9 +289,22 @@ impl Decoder for GossipsubCodec {
         let rpc = rpc_proto::Rpc::decode(&packet[..])?;
 
         let mut messages = Vec::with_capacity(rpc.publish.len());
-        for publish in rpc.publish.into_iter() {
+        for message in rpc.publish.into_iter() {
+            // verify message signatures if required
+            if !self.allow_unsigned {
```

Comment:
Do I understand correctly that when `allow_unsigned` is `true` we also allow invalid signatures?

Reply:
Correct, I've left a message about this below